internal/kafka/internal/services/kafka_tls_certificate_management/memory_storage.go

```diff
@@ -0,0 +1,115 @@
+package kafka_tls_certificate_management
+
+import (
+	"context"
+	"io/fs"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/caddyserver/certmagic"
+)
+
+type inMemoryStorageItem struct {
+	value []byte
+	mu    *sync.Mutex
+	sync.Locker
+	lastModified time.Time
+}
+
```

Comment:
It took me a while to figure it out and I am guess I would not be the only one:

This implements the `certmagic.Storage` Go interface.
Can we add interface guard clauses here to make it evident? The same should be done to the other implementations.